src/dmd/root/filename.d

```diff
@@ -631,71 +631,74 @@ nothrow:
         }
     }
 
+    /**
+       Ensure that the provided path exists
+
+       Accepts a path to either a file or a directory.
+       In the former case, the `basepath` (path to the containing directory)
+       will be checked for existence, and created if it does not exists.
+       In the later case, the directory pointed to will be checked for existence
+       and created if needed.
```

Comment:
It has nothing to do with whether the path leads to a file or a directory, as that has no effect on the behavior of the function. This detail caused me to spend some time looking at the function implementation, because adding the detail implies it has relevance. Perhaps a better description would be simply:
```
If `path` does not exist, create it.
```